File: src/main/java/com/back/nbe9112team06/domain/timeblock/entity/AvailableTime.java
package com.back.nbe9112team06.domain.timeblock.entity;

import com.back.nbe9112team06.domain.meeting.entity.Meeting;
import com.back.nbe9112team06.global.entity.BaseEntity;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalTime;

@Entity
@Getter
@NoArgsConstructor
public class AvailableTime extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "available_date_time_id")
    private AvailableDateTime availableDateTime;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "time_block_id")
    private TimeBlock timeBlock;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "meeting_id")
    private Meeting meeting;

    private LocalTime time;

    @Column(name = "created_by")
    private String createdBy;

    public static AvailableTime create(AvailableDateTime availableDateTime, TimeBlock timeBlock, Meeting meeting, LocalTime time) {
        AvailableTime availableTime = new AvailableTime();
        availableTime.availableDateTime = availableDateTime;
        availableTime.timeBlock = timeBlock;
        availableTime.meeting = meeting;
        availableTime.time = time;
        availableTime.createdBy = timeBlock.getParticipant().getGuestName();
        return availableTime;
    }

}

File: src/main/java/com/back/nbe9112team06/domain/timeblock/entity/AvailableTime.kt
package com.back.nbe9112team06.domain.timeblock.entity

import com.back.nbe9112team06.domain.meeting.entity.Meeting
import com.back.nbe9112team06.global.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import java.time.LocalTime

@Entity
class AvailableTime(

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "available_date_time_id")
    var availableDateTime: AvailableDateTime,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "time_block_id")
    var timeBlock: TimeBlock,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "meeting_id")
    var meeting: Meeting,

    var time: LocalTime,

    @Column(name = "created_by")
    var createdBy: String,

    ) : BaseEntity() {

    companion object {
        @JvmStatic
        fun create(
            availableDateTime: AvailableDateTime,
            timeBlock: TimeBlock,
            meeting: Meeting,
            time: LocalTime,
        ): AvailableTime = AvailableTime(
            availableDateTime = availableDateTime,
            timeBlock = timeBlock,
            meeting = meeting,
            time = time,
            createdBy = timeBlock.participant.guestName,
        )
    }
}